uwuno/src/main/java/com/learning/uwuno/util/playerList.java

```diff
@@ -0,0 +1,209 @@
+package com.learning.uwuno.util;
+
+import com.learning.uwuno.errors.internalServerError;
+import com.learning.uwuno.player;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.LinkedList;
+
+// Container class for a list of players
+public class playerList extends LinkedList<player> {
+    // Class Variables
+    private final String parentUID;
+    // Ensure that this HashSet and the LinkedList is kept in a consistent state
+    private HashSet<String> playerNames;
+
+    // Class functions
+    public playerList(String uid) {
+        parentUID = uid;
+        playerNames = new HashSet<String>();
+    }
+
+    // Only Call this function from within this.add(), all the checking code is there
+    private boolean addPlayerToSet(player newPlayer) {
+        // TODO: Make this more robust, there should be some way to revert back to a valid state in the future
+        // ie. super.add() can return true, but not playerNames.add() which leaves the object in an inconsistent state
+        return super.add(newPlayer) && playerNames.add(newPlayer.getName());
+    }
+
+    public player changeName(player player, String newName) {
+        String revisedName = validateAndReturnNewName(newName);
+        if (!playerNames.remove(player.getName()))
+            throw new internalServerError("Previous name does not exists in hash set");
+        playerNames.add(revisedName);
+        player.setName(revisedName);
+        return player;
+    }
+
+    // Checks name against current list of names and returns unique name (ie. adds hash to name if name already in use)
+    private String validateAndReturnNewName(String name) {
+        String newName = name;
+        if (containsName(name)) {
+            boolean isNameInUse = true;
+            while (isNameInUse) {
+                String hash = "#" + String.valueOf((Math.random() * 9999) + 1000).substring(0, 4);
+                newName = name + hash;
+                isNameInUse = containsName(newName);
+            }
+        }
+        return newName;
+    }
+
+    // Checks HashSet and not the LinkedList itself
+    public boolean containsName(String name) {
+        return playerNames.contains(name);
+    }
+
+    // Override Functions
```

Comment:
What exactly breaks if we don't extend LinkedList and why do we need to override all the functions? How would it compare if we just have a private field for list of players (e.g. LinkedList<player>)? Reimplementing/overriding all these unused functions seem a bit weird to me.

Reply:
As mentioned in chat, any function that will change the contents of the linked list won't add/remove content from the map which will break the object's consistency.